core/src/main/java/org/web3j/protocol/http/HttpService.java

```diff
@@ -70,41 +70,43 @@
 
     private OkHttpClient httpClient;
 
-    private final String url;
+    private String[] urls;
+
+    private int counter = 0;
 
-    private final boolean includeRawResponse;
+    private boolean includeRawResponse;
 
     private HashMap<String, String> headers = new HashMap<>();
 
-    public HttpService(String url, OkHttpClient httpClient, boolean includeRawResponses) {
+    public HttpService(OkHttpClient httpClient, boolean includeRawResponses, String... urls) {
```

Comment:
You will need `HttpService(OkHttpClient httpClient, boolean includeRawResponses, String url)` too